packages/eslint-plugin/src/rules/triple-slash-reference.ts

```diff
@@ -76,14 +79,29 @@ export default util.createRule<Options, MessageIds>({
     return {
       ImportDeclaration(node): void {
         if (programNode) {
-          hasMatchingReference(node.source);
+          hasMatchingReference(node.source.value);
         }
       },
       TSImportEqualsDeclaration(node): void {
         if (programNode) {
-          const source = (node.moduleReference as TSESTree.TSExternalModuleReference)
-            .expression as TSESTree.Literal;
-          hasMatchingReference(source);
+          const source = node.moduleReference;
+          let value: string | number | bigint | boolean | RegExp | null;
+
+          switch (source.type) {
+            case AST_NODE_TYPES.Identifier:
+              value = source.name;
+              break;
+
+            case AST_NODE_TYPES.TSQualifiedName:
+              value = (source.left as TSESTree.Identifier).name;
+              break;
```

Comment:
I think we should just completely ignore these cases.
There's no import source to compare against here, so doing so will likely lead to some weirdness